sfxclient/httpsink.go

```diff
@@ -126,22 +129,42 @@ func (h *HTTPSink) doBottom(ctx context.Context, f func() (io.Reader, bool, erro
 	for k, v := range h.AdditionalHeaders {
 		req.Header.Set(k, v)
 	}
+	h.setHeadersOnBottom(ctx, req, contentType, compressed)
+	resp, err := h.Client.Do(req)
+	if err != nil {
+		// According to docs, resp can be ignored since err is non-nil, so we
+		// don't have to close body.
+		return fmt.Errorf("failed to send/recieve http request: %s: %v", err, req.Header)
+	}
+
+	return h.handleResponse(resp, respValidator)
+}
+
+type xKeyContextValue string
+
+var (
+	xDebugID      xKeyContextValue = "X-Debug-Id"
+	xTracingDebug xKeyContextValue = "X-SF-Tracing-Debug"
+	xTracingID                     = "X-SF-Tracing-ID"
+)
+
+func (h *HTTPSink) setHeadersOnBottom(ctx context.Context, req *http.Request, contentType string, compressed bool) {
 	// set these below so if someone accidentally uses the same as below we wil override appropriately
 	req.Header.Set("Content-Type", contentType)
 	req.Header.Set(TokenHeaderName, h.AuthToken)
 	req.Header.Set("User-Agent", h.UserAgent)
 	req.Header.Set("Connection", "keep-alive")
+	if v := ctx.Value(xDebugID); v != nil {
+		req.Header.Set(string(xDebugID), v.(string))
+	}
+	if v := ctx.Value(xTracingDebug); v != nil {
+		hn, _ := os.Hostname()
+		req.Header.Set(string(xTracingDebug), v.(string))
+		req.Header.Set(xTracingID, hn+"-"+strconv.FormatInt(rand.Int63(), 10))
```

Comment:
how often we will be enabling Tracing-Debug?

Reply:
Very rarely, why?

Reply:
was just wondering how many time `+` will be done.